### PR TITLE
Add dynamodb utility functions from DSS.

### PR DESCRIPTION
**Pull request by DailyDreaming**
Migrating these from the DSS (where they have testing; https://github.com/HumanCellAtlas/data-store/blob/master/dss/dynamodb/__init__.py) so that fusillade and the DSS can share them.  I've made some changes so that it's a class and the table and client plug in now.

Tests should be added to this in a follow up PR.

### CODE REVIEW

File: dcplib/aws/dynamodb.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+from typing import Generator, Optional
+
+
+class DynamoDBItemNotFound(Exception):
+    pass
+
+
+class DynamoDB:
+    def __init__(self, client, table):
+        self.client = client
```
Comment (Bento007) [inline dcplib/aws/dynamodb.py]:
```suggestion
        self.client = db
```

File: dcplib/aws/dynamodb.py [hunk 2]
```diff
@@ -0,0 +1,134 @@
+from typing import Generator, Optional
+
```
Comment (Bento007) [inline dcplib/aws/dynamodb.py]: rathern then passing in the client, just import a dynamoDB client.
```suggestion
from .clients import dynamodb as db
```

File: dcplib/aws/dynamodb.py [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+
+class DynamoDBItemNotFound(Exception):
+    pass
+
+
+class DynamoDB:
+    def __init__(self, client, table):
```
Comment (Bento007) [inline dcplib/aws/dynamodb.py]:
```suggestion
    def __init__(self, table):
```

Comment (Bento007) [reply dcplib/aws/dynamodb.py]: if you keep the class then do this

File: dcplib/aws/dynamodb.py (hunk 1 above)
Comment (Bento007) [reply dcplib/aws/dynamodb.py]: if you keep the class then do this

File: dcplib/aws/dynamodb.py (hunk 2 above)
Comment (DailyDreaming) [reply dcplib/aws/dynamodb.py]: Won't this use the dcplib client rather than the dss client then?

**Pull request merged**